interfaces/builtin/avahi_observe.go

```diff
@@ -45,6 +45,10 @@ const avahiObserveBaseDeclarationSlots = `
 const avahiObservePermanentSlotAppArmor = `
 network netlink,
 
+# Allow access to deamon to create socket
+/{,var/}run/avahi-daemon/  w,
+/{,var/}run/avahi-daemon/{pid,socket} rw,
```

Comment:
Wouldn't this be better served in SNAP_COMMON or SNAP_DATA?

Reply:
SNAP_COMMON is where socket is created at the moment, when avahi is installed as snap. Problem is that on classic system when avahi is deployed through deb packages, socket is created in /run/avahi-daemon which creates several problems.
Client needs to worry about two location, libnss_mdns.so has path hardcoded and compiled in, so one would need to handle two binary versions in one snap. Also on classic system there is no content interface so client will still need permission /run/avahi-daemon/socket rw
So cleanest way seems to be to unify between core and classic